BibLaTeX.js

```diff
@@ -478,11 +483,8 @@ function encodeFilePathComponent(value) {
 
 			if (!type) type = "misc";
 
-			var citekey = "";
-			if (!citekey) {
-				// create a unique citation key
-				citekey = buildCiteKey(item, citekeys);
-			}
+			var citekey = buildCiteKey(item, extraFields, citekeys);
+			var extraFields = item.extra ? parseExtraFields(item.extra) : null;
```

Comment:
I always get confused in how JS handles variables, but for clarity we should define the variable before we use it, i.e. flip these two lines.

Reply:
Oops. Change submitted. That was unintentional.

Reply:
I've also bumped the dates.